x-pack/plugins/ai_infra/product_doc_base/server/tasks/install_all.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type {
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import type { InternalServices } from '../types';
+import { isTaskCurrentlyRunningError } from './utils';
+
+export const INSTALL_ALL_TASK_TYPE = 'ProductDocBase:InstallAll';
+export const INSTALL_ALL_TASK_ID = 'ProductDocBase:InstallAll';
+
+export const registerInstallAllTaskDefinition = ({
+  getServices,
+  taskManager,
+}: {
+  getServices: () => InternalServices;
+  taskManager: TaskManagerSetupContract;
+}) => {
+  taskManager.registerTaskDefinitions({
+    [INSTALL_ALL_TASK_TYPE]: {
+      title: 'Install all product documentation artifacts',
+      timeout: '10m',
+      maxAttempts: 3,
```

Comment:
We should check with response ops about the `cost` attribute for all of our tasks (https://github.com/pgayvallet/kibana/blob/1ddad2f6cd6b23333a6d2070e50d705ec79ab2fd/x-pack/plugins/task_manager/server/task.ts#L21-L25).

We know this is a long-running task which is capable of emitting event loop utilization warnings. Right now our `cost` options are `tiny`, `normal`, and `extra large`. I'm not sure if this qualifies as `extra large` or not, but maybe response ops can help us determine the appropriate `cost` to select.

Reply:
Good point. I asked Mike about this

Reply:
Extra large means it needs a lot of CPU and memory resources compared to other typical background tasks (5x more than normal tasks). For example, indicator match rules in security solution.

Setting extra large indicates to task manager that it should consider these types of tasks are the equivalent of five normal background tasks, so it reduces overall task concurrency to accommodate these tasks. In general we recommend optimizing the task type for memory and CPU instead of setting extra large as it does affect Kibana throughput on a given node quite a bit when they run.